week3/homework/src/manipulate.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const fs = require('fs');
+const storage = './todos.json';
+
+function runTheFunction(func, param1, param2) {
```

Comment:
You can refactor this into `readData()` function, that only reads the data and returns it as JSON.
You can then call that function from for example an `add(item)` function (and all other functions):
```
function add(item) {
    const data = readData();
    const itemObj = {
        index: toDoList.length + 1,
        description: item
    };
    toDoList.push(itemObj);

    fs.writeFile(storage, JSON.stringify(toDoList), (error) => {
        if (error) { console.log(error); }
    });
}
```
In that way you can call the `add(item)` function directly from your route, so you would have in `src/index.js`:
```
app.post('todos, (request, response) => {
    manipulate.add(request.body.task);
    response
        .status(201)
        .json({ result: "your task is added!" });
});
```